### PR TITLE
chore: convert a first batch of files to typescript

### PR DESCRIPTION
**Pull request by lostfictions**
#### Relevant info and issue/PR links 
 
Issues should be fixed, if tests pass should be ok to merge pending review

#### Testing instructions  

----
### Repository owner

- [ ] Was tested and can be merged.
- [ ] PR name follows conventional changelog format.

[PR guideline](https://github.com/PeerioTechnologies/peerio-icebear/blob/dev/docs/CONTRIBUTING.md)

### CODE REVIEW

File: src/__sdk.js [hunk 1]
```diff
@@ -1 +1 @@
-module.exports = '5.13.0';
+export default '5.13.0';
```
Comment (anri-asaturov) [inline src/__sdk.js]: this file is autogenerated in prostbump.js

**Comment by anri-asaturov** [comment]
I can give you more permissions so you can merge ts branch with dev after every small success step, how about that?

**Comment by lostfictions** [comment]
Reverted some changes, so this PR is only concerned with the conversion of the `crypto/` directory (plus changes needed for tests to actually consume Typescript files.)

File: src/crypto/index.ts [hunk 2]
```diff
@@ -0,0 +1,8 @@
+import * as cryptoUtil from './util';
+import * as keys from './keys';
+import * as publicCrypto from './public';
+import * as secret from './secret';
+import * as sign from './sign';
+import { setScrypt } from './scrypt-proxy';
+
+export { cryptoUtil, keys, publicCrypto, secret, sign, setScrypt };
```
Comment (anri-asaturov) [inline src/crypto/index.ts]: why doesn't it follow renames? too many changes?

**Comment by anri-asaturov** [comment]
It's hard to review when it doesn't follow renamed files :(

**Comment by lostfictions** [comment]
Yeah, sorry, it's not ideal. You'll notice that some of them are tracked and some aren't; there's apparently no way to manually force Git to regard an addition-removal pair as a rename, it's just by difference %.

I think the only way to 100% ensure that it continues to track all files is to rename them, immediately commit without making any changes to the files, and then change the files and make an additional commit on top (and even then I'm not 100% what would happen when squashing in this scenario :( ) I'll definitely do it this way moving forward, but can't guarantee that when a PR is squash-merged into dev the tracking won't get cut off again (eg. if we later want to use `blame`). We'll just have to see what happens.

**Comment by anri-asaturov** [comment]
```   LOG:24:36.443: ⬆️ OUT MSG: 4252["/auth/user/lookup",{"string":["r_ay7fh2ut26q8kk"]}] 
       LOG:24:36.455: ⬇️ IN MSG: 62-52[[[{"matchFields":["username"],"profile":{"encryptionPublicKey":{"_placeholder":true,"num":0},"signingPublicKey":{"_placeholder":true,"num":1},"username":"r_ay7fh2ut26q8kk","addresses":[],"hasAvatar":false,"appLabel":"peerio","firstName":"Firstname","lastName":"Lastname","urlSalt":"PCOKWVRKKWZFEKUKJ6WDIT6LGAVC2MJSFO2MRTSW2JI4VS6PJNHAKSUVESD3V6B37K4AIJNFDATLVG7HEQLPTZCHUXN5UKWVAIADDVAS2A","isDeleted":false,"props":{},"profileVersion":1}}]]] 
       WARN:24:36.468: Unhandled rejection TypeError: Cannot read property 'encryptionPublicKey' of undefined
           at /home/circleci/repo/src/models/contacts/tofu-store.js:91:21
           at Generator.next (<anonymous>)
           at Generator.tryCatcher (/home/circleci/repo/node_modules/bluebird/js/release/util.js:16:23)
           at PromiseSpawn._promiseFulfilled (/home/circleci/repo/node_modules/bluebird/js/release/generators.js:97:49)
           at Promise._settlePromise (/home/circleci/repo/node_modules/bluebird/js/release/promise.js:574:26)
           at Promise._settlePromise0 (/home/circleci/repo/node_modules/bluebird/js/release/promise.js:614:10)
           at Promise._settlePromises (/home/circleci/repo/node_modules/bluebird/js/release/promise.js:693:18)
           at Async._drainQueue (/home/circleci/repo/node_modules/bluebird/js/release/async.js:133:16)
           at Async._drainQueues (/home/circleci/repo/node_modules/bluebird/js/release/async.js:143:10)
           at Immediate.Async.drainQueues [as _onImmediate] (/home/circleci/repo/node_modules/bluebird/js/release/async.js:17:14)
           at runCallback (timers.js:763:18)
           at tryOnImmediate (timers.js:734:5)
           at processImmediate (timers.js:716:5)```

**Comment by anri-asaturov** [comment]
probably not related to this branch

**Comment by anri-asaturov** [comment]
yeah it's not, fixed in dev

**Comment by anri-asaturov** [comment]
tests pass, tell me when you want this merged

**Comment by lostfictions** [comment]
not sure if anyone else wants to/is able to do a quick review, but otherwise should be ok to merge?

**Comment by ebilg** [comment]
please hold off on merge until after this week's release

**Comment by lostfictions** [comment]
Sounds good. For now I'll make another branch off this one for further conversion work and then rebase later if necessary

**Pull request merged**